### PR TITLE
BENCH-242 Menu Page update

### PR DESCRIPTION
**Pull request by AudriusPutys**
menu_page2

### CODE REVIEW

File: src/components/LocalOrderDetails/LocalOrderDetails.tsx [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
     <div>
-      <div>
+      <div className="max-h-[17rem] overflow-auto">
         {localOrder.lineItems?.length > 0 ? (
           localOrder.lineItems.map((lineItem) => (
             <div key={lineItem.product.id} className="text-[16px]">
-              <div className="mb-3 mt-3 grid grid-cols-6" key={lineItem.product.id}>
-                <div className="col-span-2">
+              <div className="mb-3 mt-3 grid grid-cols-12" key={lineItem.product.id}>
+                <div className="text-poppins col-span-1 flex items-center justify-center rounded bg-gray-200 text-sm">{lineItem.quantity}</div>
```
Comment (harrystead2) [inline src/components/LocalOrderDetails/LocalOrderDetails.tsx]: font-poppins, not text-poppins

File: src/components/MenuItems/MenuItems.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ export const MenuItems = ({ products }: Props): ReactElement => {
   }
   return (
     <div>
-      <div className="mb-12 grid grid-cols-3 px-16">
+      <div className="mb-12 mr-10 grid max-h-[60vh] grid-cols-3 overflow-auto px-16">
```
Comment (harrystead2) [inline src/components/MenuItems/MenuItems.tsx]: Try to use precise measurements e.g mb-[20px], don't have to always if its like mb-1 or mt-2, but it is easier to read for larger measurements.

File: src/components/OrderType/OrderType.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const [activeButton, setActiveButton] = useState<number>(1);
+
+  return (
+    <div>
+      <h1 className="font-poppins mt-2 text-center text-xl font-semibold">Order Summary</h1>
+      <div className="mt-3 mb-4 grid h-8 grid-cols-2 rounded-[14px] bg-gray-200 align-bottom">
+        <div className="col-span-1 ml-1 pt-1">
+          <button
+            onClick={() => setActiveButton(1)}
+            className={`font-poppins rounded-xl ${activeButton == 1 ? 'bg-[#ffffff]' : ''} px-4 text-[#333F4C] hover:cursor-pointer hover:bg-gray-50`}
```
Comment (harrystead2) [inline src/components/OrderType/OrderType.tsx]: If you rebase, font-poppins should be applied everywhere now so no need to use it again and again in classes. Would only need to change font weight and style individually.

File: src/pages/Menu/Menu.tsx [hunk 4]
```diff
@@ -30,17 +30,20 @@ export const MenuPage = (): ReactElement => {
...
+          </div>
+          <div className="font-poly mb-6 text-lg italic text-gray-300">Served on a brioche bun with house fries and fresh salad</div>
           <MenuItems
             products={products.data.filter(
               (product) => product.retired === false && product.categories?.find((categoryId) => categoryId === selectedCategory)
             )}
           />
         </div>
-        <div className="col-span-3">
+        <div className="col-span-3 mt-32">
```
Comment (harrystead2) [inline src/pages/Menu/Menu.tsx]: Would probably make this a percentage - e.g mb-[something%], or add the panel component alongside the cards.

**Review by harrystead2 — CHANGES_REQUESTED** [review]

**Comment by harrystead2** [comment]
### Styling Changes: 

### General
1. Black or dark colour is #333F4C, not fully black. 
2. Padding and margin not quite right in most places. 
3. Fonts wrong in most places, including Menu title. 

### Tabs
1. When you click on Sides tab, burgers appear. But nothing appears in burgers
   tab at the start - why? Is that a problem with the filter on frontend or did you add
   products called burger to sides on backend?
2. change font-thin to font-[300] on text in tabs.
3. change text size on tabs to be 16px.
4. each tab should have a width of 72px and height of 32px,  
5. distance between tabs should be 40px. 

### Cards
1. Padding is wrong - between picture and title, between title and description, between 
description and button, between button and bottom of card.
2. Cards look too wide and not long enough - should be 247.33px wide and 365px high.
3. Pictures looks stretched out? Probably because of 2. But if still stretched out 
   there's other things to use to get it looking better.
4. No 'new' or 'recommended' tag in top left of one of the cards. Not sure if this will 
   come from backend in future, but would be nice to manually add it to one of the 
   cards for now. 
5. There shouldn't be a scroll bar between cards and panel. Should only be a scroll
   bar on far right at end of screen. 
6. Font is wrong on description. Shouldn't be bold. Might need to get the italicised
   poly font from google. Should be font-[400], that's if we have the poly 400 italicised
   version, if not need to pull from google fonts. 
7. Font is wrong on buttons. Should not be fully bold. Should be - font-[400].
8. The 'Served on a brioche bun' subheading should be fully white, not greyed out. 

### Panel
1. No need for delivery/collection tab on panel. 
2. Use Bin component I made for trash icon (rebase to develop to get it).
3. Shouldn't be a line underneath last item in panel - use this - [&:not(:last-child)]:border-b
4. Icon sizes for quantity and trash not quite right - should be 30px high and 30px wide. 
5. Should be no scroll bar on bar - should grow as items are added. Remove overflow and add 'grow'.
6. Service charge ps are too big and should not be bold. 
7. No need for delivery fee - remove. 
8. Padding not right between service charge and total. 
9. Total font size too big - should be 20px. 
10. Margin between back to start screen and allergen board button not quite right, and should also
     be slightly bigger than panel on both sides. 
11. Font not right either on buttons below panel, should be - text-14px, font-poppins, font-[400], leading-[21px], text- 
      [#A2AAB6] 
12. Panel itself is not correct size, should be width: 357px; height: 618px;
13. Panel should be closer to middle. 
14. Top of panel should be in line with the divider between items and category names. 
15. Padding between quantity/trash icon and name/price is wrong.

**Comment by harrystead2** [comment]
I'll look into the code smell, might have to make the arrow function a normal function for now.

**Comment by harrystead2** [comment]
> I'll look into the code smell, might have to make the arrow function a normal function for now.

Yeah for performance, address the smell and change to a normal function

File: src/components/OrderCreateForm/OrderCreateForm.tsx [hunk 5]
```diff
@@ -6,11 +6,13 @@ import { useNavigate } from 'react-router-dom';
 import { RouteConstants } from 'utilities/route-constants';
 import { CreatedOrderDto } from 'dtos/Order/CreatedOrderDto';
 import { useLocalOrder } from 'context/OrderContext';
+import { GBPFormat } from 'utilities/GBPFormat';
 
 export const OrderCreateForm = (): ReactElement => {
   const { order, createOrder, updateOrder } = useOrder();
   const { localOrder, setOrderId } = useLocalOrder();
   const navigate = useNavigate();
+  const noProductsInBasket = localOrder.lineItems?.length === 0;
```
Comment (harrystead2) [inline src/components/OrderCreateForm/OrderCreateForm.tsx:15]: rename this to - lineItemsExist

**Review by harrystead2 — DISMISSED** [review]

File: src/pages/Menu/Menu.tsx [hunk 6]
```diff
@@ -24,19 +24,23 @@ export const MenuPage = (): ReactElement => {
...
-      <div className="mt-10 grid grid-cols-12">
-        <div className="col-span-1"></div>
-        <div className="col-span-8 text-center">
+      <div className="grid grid-cols-12">
+        <div className="col-span-2"></div>
+        <div className="col-span-6 pl-[100px] text-center">
+          <div className="mb-5">
+            <MenuCategories selectedCategory={selectedCategory} setSelectedCategory={setSelectedCategory} categories={categories.data} />
+          </div>
+          <div className="font-poly mb-6 text-lg italic text-[#E4EAEB]">Served on a brioche bun with house fries and fresh salad</div>
```
Comment (bethcryer) [inline src/pages/Menu/Menu.tsx]: Should this be using the selected category's `description` field?

**Review by bethcryer — CHANGES_REQUESTED** [review]

File: src/pages/Menu/Menu.tsx (hunk 6 above)
Comment (harrystead2) [reply src/pages/Menu/Menu.tsx]: Ooo yes

Comment (AudriusPutys) [reply src/pages/Menu/Menu.tsx]: good catch, updated now

**Review by harrystead2 — APPROVED** [review]

**Review by bethcryer — APPROVED** [review]

**Pull request merged**